packages/roosterjs-content-model/lib/publicApi/utils/selectWord.ts

```diff
@@ -0,0 +1,103 @@
+import { ContentModelBlockGroup } from '../../publicTypes/group/ContentModelBlockGroup';
+import { ContentModelSegment } from '../../publicTypes/segment/ContentModelSegment';
+import { ContentModelText } from '../../publicTypes/segment/ContentModelText';
+import { createText } from '../../modelApi/creators/createText';
+
+export function selectWord(
+    path: ContentModelBlockGroup[],
+    marker: ContentModelSegment
+): ContentModelSegment[] {
+    const parent = path[0];
+    const block = parent.blocks[0];
+
+    if (block.blockType == 'Paragraph') {
+        const segments: ContentModelSegment[] = [];
+        let markerSelectionIndex = block.segments.indexOf(marker);
+        for (let i = markerSelectionIndex - 1; i >= 0; i--) {
+            const currentSegment = block.segments[i];
+            if (currentSegment.segmentType == 'Text') {
+                const found = findDelimiter(currentSegment, false);
+                if (found != null) {
+                    if (found == currentSegment.text.length) {
+                        break;
+                    }
+                    splitTextSegment(block.segments as ContentModelText[], i, found);
```

Comment:
Although this work, but we should not do this kind of type casting, because not all segments are Text node here.

Instead you can pass the array and the found text node rather than the index, so no need to do type casting any more:

```ts
function splitTextSegments(segments: ContentModelSegment[], textToSplit: ContentModelText, ....) {
   const index = segments.indexOf(textToSplit);
   if (index >= 0) {
      ... 
   }
}
```

Reply:
Why do you recommend I do indexOf when I already have the index?

Reply:
Understand. This is just to avoid a type casting.

Do indexOf() twice will cost more time, but it is not that bad. This operation it self is super fast. But unnecessary type casting can hide bugs later when we change some type.